dds/DCPS/DataSampleHeader.cpp

```diff
@@ -34,7 +34,7 @@ namespace {
       ACE_Message_Block temp(mb.data_block (), ACE_Message_Block::DONT_DELETE);
       temp.rd_ptr(mb.rd_ptr()+offset);
       temp.wr_ptr(mb.wr_ptr());
-      OpenDDS::DCPS::Serializer ser(&temp, swap);
+      OpenDDS::DCPS::Serializer ser(&temp, OpenDDS::DCPS::Encoding::KIND_CDR_UNALIGNED, swap ? OpenDDS::DCPS::ENDIAN_LITTLE : OpenDDS::DCPS::ENDIAN_BIG);
```

Comment:
I think you're mixing up NATIVE/NONNATIVE with BIG/LITTLE here and looks like you've done this at least a couple of times. Also this line and the changed line below are way too long.

For this particular situation these `mb_*` functions are only used in `DataSampleHeader::partial` and it looks like there is already an encoding object that I put there, but it's not doing everything it could do. You could move that encoding object up, the swap argument on these functions `mb_*` functions to `const Encoding& encoding` and pass the encoding object in, and set swap when it has that information.

Reply:
You are correct. I have changed the ? x : y to the proper ones.  I got the swap bytes ones mixed up with byte order ones.

I will now look into factoring out encoding where possible.

Reply:
I made an encoding object for each file where no local if statement was included and a specific encoding setup was used more than once in that file.